hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/contracts/operation/HederaSStoreOperation.java

```diff
@@ -58,13 +60,15 @@ public OperationResult execute(final MessageFrame frame, final EVM evm) {
         final var value = UInt256.fromBytes(frame.popStackItem());
         final var addressOrAlias = frame.getRecipientAddress();
         final var account = frame.getWorldUpdater().getAccount(addressOrAlias).getMutable();
+        final Supplier<UInt256> oldValue = Suppliers.memoize(() -> account.getStorageValue(key));
+        final Supplier<UInt256> originalValue = Suppliers.memoize(() -> account.getOriginalStorageValue(key));
         if (account == null) {
             return ILLEGAL_STATE_CHANGE_RESULT;
         }
         final var address = account.getAddress();
         final var slotIsWarm = frame.warmUpStorage(address, key);
         final var calculator = gasCalculator();
-        final var calcGasCost = calculator.calculateStorageCost(account, key, value)
+        final var calcGasCost = calculator.calculateStorageCost(value, oldValue, originalValue)
```

Comment:
Maybe some comments to inform how we're to understand the 3 separate but similarly name variables.

Reply:
The API from besu changed.  It's explained in the besu code already.